### ISSUE TITLE
Keytrack behaves differently when mod-mapped via saw modulator

### ISSUE THREAD

**Issue by Andreya-Autumn** [surge-synthesizer/surge#6917]
[Modulo.fxp.zip](https://github.com/surge-synthesizer/surge/files/11183472/Modulo.fxp.zip)

Load that

Learn that scene A has keytracking assigned to a timbre param, and scene B has the exact same assignment via the stopped saw LFO phase mod-mapping trick.
Thus: expect to hear the same behavior on both scenes.

Play a scale from C4 up to C6 in Scene A
Switch to Scene B and play the same scale again
Ponder why those behave differently
Proft?!!

**Comment by Andreya-Autumn** [comment]
@baconpaul I tried switching which scene has the mod-map, and the weird octave-modulo behavior followed.

**Comment by mkruselj** [comment]
This sounds like a possibly breaking change would have to be introduced to fix the behavior?

Should probably move it to XT2.0 milestone then...

**Comment by Andreya-Autumn** [comment]
You're right of course. It seems pretty plausible to me that no one ever used this particular broken behavior for patches or whatever. Though obviously we can't know that for sure.

I wonder, is it possible to do streaming trickery like with the other "legacy" behaviors that remain, but without actually exposing the legacy behavior in the UI? So old saved projects/patches load with the borked behavior, but new initialized instances have it fixed. Should be possible right? Bit un-kosher perhaps. But I bet it'd actually work here since this is such an edgy edge case. 

Of course, all of that is dependant on actually having a fix. 🙃 But you know.

**Comment by mkruselj** [comment]
Calculating modulation is one of our tightest processing loops after the DSP stuff, so doing queries based on version in that loop might be detrimental to performance. @baconpaul will correct me if I'm wrong.

I think it's just much safer to do it properly in XT2.

**Comment by baconpaul** [comment]
i didn't try the patch at a keyboard but it seems to me that this results from key track being one of the modulation sources which has values outside +/- 1 so if you map it to phase of a periodic saw the saw periodicity will wrap

That is, if you take a function which ranges from, say, -5 to 5 (which is the keyboard range from 0 to 120 with key track of 60) and then transform it with a function which is a saw, so periodically push it into -1,1, it will sound different.

Is that what's happening here? If so that's not a bug.

**Comment by mkruselj** [comment]
Yeah that sounds like what's happening here.
